driver-core/src/test/java/com/datastax/driver/core/CCMBridge.java

```diff
@@ -1048,6 +1047,34 @@ public CCMBridge build() {
         cassandraVersion = this.version;
       }
 
+      selectedPorts.clear();
+      // Preemptively banning every number that looks like explicitly provided port
+      for (Map.Entry<String, Object> entry : this.cassandraConfiguration.entrySet()) {
+        Object value = entry.getValue();
+        try {
+          Integer result = Integer.parseInt(value.toString());
+          if ((1 <= result) && (result <= 65535)) {
+            selectedPorts.add(result);
+          }
+        } catch (NumberFormatException e) {
+        }
+      }
+      if (dseVersion != null) {
+        for (Map.Entry<String, Object> entry : this.dseConfiguration.entrySet()) {
```

Comment:
It seems that you could change these two lines to `for (Object value : this.dseConfiguration.values()) {`, no?